tests/driver/runner_test.py

```diff
@@ -4913,6 +4913,90 @@ async def test_delete_ilm_policy_without_request_params(self, es):
 
 
+class TestSqlRunner:
+    default_response = {
+        "columns": [{"name": "first_name", "type": "text"}],
+        "rows": [
+            ["Georgi"],
+            ["Bezalel"],
+        ],
+        "cursor": "firstCursor",
+    }
+
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_fetch_one_page(self, es):
+        es.transport.perform_request = mock.AsyncMock(return_value=io.StringIO(json.dumps(self.default_response)))
+
+        sql_runner = runner.Sql()
+        params = {
+            "operation-type": "sql",
+            "body": {
+                "query": "SELECT first_name FROM emp",
+                "fetch_size": 2,
+                "page_timeout": "10s",
+                "request_timeout": "20s",
+            },
+        }
+
+        async with sql_runner:
+            result = await sql_runner(es, params)
+
+        assert result == {"success": True, "weight": 1, "unit": "ops"}
+
+        es.transport.perform_request.assert_awaited_once_with("POST", "/_sql", body=params["body"])
+
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_fetch_all_pages(self, es):
+        es.transport.perform_request = mock.AsyncMock(return_value=io.StringIO(json.dumps(self.default_response)))
+
+        sql_runner = runner.Sql()
+        params = {"operation-type": "sql", "body": {"query": "SELECT first_name FROM emp"}, "pages": 3}
+
+        async with sql_runner:
+            result = await sql_runner(es, params)
+
+        assert result == {"success": True, "weight": 3, "unit": "ops"}
+
+        es.transport.perform_request.assert_has_calls(
+            [
+                mock.call("POST", "/_sql", body=params["body"]),
+                mock.call("POST", "/_sql", body={"cursor": self.default_response["cursor"]}),
+                mock.call("POST", "/_sql", body={"cursor": self.default_response["cursor"]}),
+            ]
+        )
+
+    @mock.patch("elasticsearch.Elasticsearch")
+    @run_async
+    async def test_failure_on_too_few_pages(self, es):
+        es.transport.perform_request = mock.AsyncMock(
+            side_effect=[io.StringIO(json.dumps(self.default_response)), io.StringIO(json.dumps({"rows": [["John"]]}))]
+        )
+
+        sql_runner = runner.Sql()
+        params = {
+            "operation-type": "sql",
+            "body": {
+                "query": "SELECT first_name FROM emp",
+            },
+            "pages": 3,
+        }
+
+        with pytest.raises(exceptions.DataError) as ctx:
+            async with sql_runner:
+                await sql_runner(es, params)
+
+        assert ctx.value.message == "Result set has been exhausted before all pages have been fetched, 1 page(s) remaining."
+
+        es.transport.perform_request.assert_has_calls(
+            [
+                mock.call("POST", "/_sql", body=params["body"]),
+                mock.call("POST", "/_sql", body={"cursor": self.default_response["cursor"]}),
+            ]
+        )
+
+
```

Comment:
We should add a test that ensures that an exception is raised if the `body` and `body.query` parameters aren't included. Something like this would do it:

```suggestion
    @mock.patch("elasticsearch.Elasticsearch")
    @run_async
    async def test_mandatory_body_param(self, es):
        sql_runner = runner.Sql()
        params = {
            "operation-type": "sql",
            "pages": 3,
        }

        with pytest.raises(exceptions.DataError) as exc:
            await sql_runner(es, params)
        assert exc.value.args[0] == (
            "Parameter source for operation 'sql' did not provide the mandatory parameter 'body'. "
            "Add it to your parameter source and try again."
        )
        
    @mock.patch("elasticsearch.Elasticsearch")
    @run_async
    async def test_mandatory_query_in_body_param(self, es):
        sql_runner = runner.Sql()
        params = {
            "operation-type": "sql",
            "body": {},
            "pages": 3,
        }

        with pytest.raises(exceptions.DataError) as exc:
            await sql_runner(es, params)
        assert exc.value.args[0] == (
            "Parameter source for operation 'sql' did not provide the mandatory parameter 'body.query'. "
            "Add it to your parameter source and try again."
        )
```